docs/content/docs/testing/testing.md

```diff
@@ -0,0 +1,58 @@
+
+This document will explain how to use these tools to get started with writing automation for Fluid applications against a service.  It will focus on interactions with the service rather than automation in general, and will not cover the automation tools themselves or scenarios that do not require a service.
+
+{{% callout important %}}
+
+Specific examples of third-party libraries are used throughout this document for certain tasks.  While these are the same libraries used internally by Fluid, you are free to accomplish these tasks in other ways.  Microsoft is not responsible for [liability statement?].
+
+{{% /callout %}}
+
+## Automation against Tinylicious
+
+Automation against Tinylicious is useful for scenarios such as merge validation which want to be unaffected by service interruptions.  Your automation should be responsible for starting a local instance of Tinylicious along with terminating it once tests have completed.  One way you can do this is by using the `start-server-and-test` package in the following npm scripts.
+
+```json
+"scripts": {
+    ...
+    "start:tinylicious": "tinylicious > tinylicious.log 2>&1",
+    "test:mocha": "mocha",
+    "test:tinylicious": "start-server-and-test start:tinylicious 7070 test:mocha",
+    ...
+}
+```
+
+Running the `test:tinylicious` script will start Tinylicious, wait until port 7070 responds (the default port on which Tinylicious runs), run the test script, and then terminate Tinylicious.  Your tests can then use `TinyliciousClient` as usual (see [Tinylicious]({{< relref tinylicious.md >}})).
+
+## Automation against Azure Fluid Relay
+
+With a test tenant for Azure Fluid Relay, your automation can connect to it in the same way as your production tenant and needs only provide the appropriate connection configuration.  See [Connect to an Azure Fluid Relay service]({{< relref azure-frs.md >}}) for more details.
+
+### Azure Fluid Relay as an abstraction for Tinylicious
+
+The Azure Fluid Relay client can also connect to a local Tinylicious instance.  This allows you to use a single client type between tests against live and local service instances, where the only difference is the configuration used to create the client.
+
+```javascript
+const user = {
+    id: "UserId",
+    name: "Test User",
+};
+const config = {
+    tenantId: "local",
+    tokenProvider: new InsecureTokenProvider("fooBar", user),
+    orderer: "http://localhost:7070",
+    storage: "http://localhost:7070",
+};
+// This FrsClient instance connects to a local Tinylicious
+// instance rather than a live Azure Fluid Relay service
+const client = new FrsClient(config);
+```
+
+These values for `tenantId`, `orderer`, and `storage` correspond to those for Tinylicious, where `7070` is the default port for Tinylicious.
```

Comment:
This maybe for a follow-up PR but could we also include a code sample here for a test that runs against the FRS client? Perhaps it could be a walkthrough of this test from @timtwang 's PR to step a developer through the code? https://github.com/microsoft/FluidFramework/pull/6761/files#diff-498c7acedc92c2216986033b7ea00bf459dd0c8dcd82221b3c36d649b22576f5

Reply:
Which part of the test do you think would benefit from the walkthrough?  I figured the above sample covers the frs-specific testing stuff, so it seemed to me all we would be adding is a trivial test which doesn't cover new fluid-specific concepts.  I agree a walkthrough of testing for a more complex app built on frs-client might be useful to have somewhere, but it also feels like to me it's not our responsibility to teach general testing :/

Reply:
If we go with this route, perhaps it's a good idea to have code sample walk-through for both FRS Client and Tinylicious Client? I think this helps developers see the patterns between 2 clients.

Reply:
@heliocliu I was thinking that it may be useful to show that the same client can be used across multiple tests if the configuration stays the same, as well as that the new document ID can get generated in the `beforeEach` section. While these may differ for the developer's own tests based off of their needs, having that guidance in place might be helpful to get a starting example running and then the developer can tailor it further to meet there specific reqs. I agree that we shouldn't go all in to teach them general testing practices, but having a basic example of how Fluid could be integrated with those general practices could be a good starting foundation